service/src/main/kotlin/fi/espoo/evaka/vasu/VasuController.kt

```diff
@@ -389,12 +393,46 @@ class VasuController(
                     tx.freezeVasuPlacements(id)
                 }
 
-                events.forEach { eventType ->
-                    tx.insertVasuDocumentEvent(
-                        documentId = id,
-                        eventType = eventType,
-                        employeeId = user.id
-                    )
+                val addedEvents =
+                    events.map { eventType ->
+                        tx.insertVasuDocumentEvent(
+                            documentId = id,
+                            eventType = eventType,
+                            employeeId = user.id
+                        )
+                    }
+
+                if (events.contains(MOVED_TO_CLOSED)) {
+                    val duplicates = tx.listPersonByDuplicateOf(document.basics.child.id)
+                    if (duplicates.isNotEmpty()) {
+                        val child = tx.getChild(document.basics.child.id)!!
+                        val template = tx.getVasuTemplate(document.templateId)!!
+                        val allEvents = document.events + addedEvents
+                        duplicates.forEach { duplicate ->
+                            val duplicateChild =
+                                tx.getChild(duplicate.id)
+                                    ?: tx.createChild(child.copy(id = duplicate.id))
```

Comment:
I'm probably missing something here, but why do we need to create a `child` for a duplicated person here? Are these not created when the person is duplicated?

Or is this just being defensive for a potential case of bad data?

Reply:
The latter.